src/transforms/mod.rs

```diff
@@ -54,6 +54,25 @@ enum BuildError {
     InvalidSubstring { name: String },
 }
 
+/// The user configuration to choose the metric tag strategy.
+#[configurable_component]
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+#[serde(rename_all = "snake_case")]
+pub enum MetricTagsValues {
+    /// Tag values will be exposed as single strings, the
+    /// same as they were before this config option. Tags with multiple values will show the last assigned value, and null values
+    /// will be ignored.
+    Single,
+    /// All tags will be exposed as arrays of either string or null values.
+    Full,
+}
+
+impl Default for MetricTagsValues {
+    fn default() -> Self {
+        Self::Single
+    }
+}
```

Comment:
I think the default value can now be handled with `#[derive(Default)]`:
```suggestion
#[configurable_component]
#[derive(Copy, Clone, Debug, Default, PartialEq, Eq)]
#[serde(rename_all = "snake_case")]
pub enum MetricTagsValues {
    /// Tag values will be exposed as single strings, the
    /// same as they were before this config option. Tags with multiple values will show the last assigned value, and null values
    /// will be ignored.
    #[default]
    Single,
    /// All tags will be exposed as arrays of either string or null values.
    Full,
}
```